jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/JMSConnectionFactory.java

```diff
@@ -338,7 +338,12 @@ public Connection createConnection(String userName, String password) throws JMSE
                     connection = tConFac.createTopicConnection(userName, password);
                 }
                 if (isDurable && !isSharedSubscription) {
-                    connection.setClientID(clientId);
+                    if (connection != null) {
```

Comment:
Can we do like this? 

First check whether connection is null and throw error from there. Else, call the setClientID method ?

Basically, reverse the if condition and remove the else part

Reply:
This is fixed.